ui/StatusQ/src/wallet/tokendata.cpp

```diff
@@ -0,0 +1,102 @@
+#include "tokendata.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QList>
+
+CollectiblePreferencesItemType tokenDataToCollectiblePreferencesItemType(const TokenData& token)
+{
+    if (!token.communityId.isEmpty()) {
+        return CollectiblePreferencesItemType::CommunityCollectible;
+    } else if (!token.collectionUid.isEmpty()) {
+        return CollectiblePreferencesItemType::Collection;
+    } else {
+        return CollectiblePreferencesItemType::NonCommunityCollectible;
+    }
+    // TODO #13313: When is CollectiblePreferencesItemType::Community used? instead of CommunityCollectible?
+}
+
+TokenOrder::TokenOrder() : sortOrder(undefinedTokenOrder) {}
+
+TokenOrder::TokenOrder(const QString& symbol,
+                       int sortOrder,
+                       bool visible,
+                       const QString& communityId,
+                       const QString& collectionUid,
+                       CollectiblePreferencesItemType type)
+    : symbol(symbol)
+    , sortOrder(sortOrder)
+    , visible(visible)
+    , communityId(communityId)
+    , collectionUid(collectionUid)
+    , type(type)
+{
+}
+
+/// reverse of tokenOrdersFromJson
+QString tokenOrdersToJson(const SerializedTokenData& dataList, bool areCollectible)
+{
+    QJsonArray jsonArray;
+    for (const TokenOrder& data : dataList) {
+        QJsonObject obj;
+        obj["key"] = data.symbol;
+        obj["position"] = data.sortOrder;
+        obj["visible"] = data.visible;
+        if (!data.collectionUid.isEmpty()) {
+            obj["collectionUid"] = data.collectionUid;
+        }
+        if (!data.communityId.isEmpty()) {
+            obj["communityId"] = data.communityId;
+        }
+
+        if (areCollectible) {
+            // see CollectiblePreferences in src/backend/collectibles_types.nim
+            obj["type"] = static_cast<int>(data.type);
+        } else { // is asset
+            // see TokenPreferences in src/backend/backend.nim
+            // TODO #13312: handle "groupPosition" for asset
```

Comment:
This is fixed in my PR here: https://github.com/status-im/status-desktop/pull/13984

Reply:
I will take into consideration when tackling assets next iteration